.pre-commit-config.yaml

```diff
@@ -0,0 +1,35 @@
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v5.0.0
+    hooks:
+      - id: trailing-whitespace
+        types: [python]
+      - id: check-case-conflict
+      - id: check-docstring-first
+      - id: check-merge-conflict
+      - id: check-yaml
+      - id: debug-statements
+      - id: detect-private-key
+      - id: name-tests-test
+      - id: end-of-file-fixer
+        types: [python]
+      - id: requirements-txt-fixer
+
+  - repo: https://github.com/pycqa/isort
+    rev: 5.13.2
+    hooks:
+      - id: isort
+        name: isort (python)
+        exclude: >-
+          (?x)^(
+          .*__init__\.py$
+          )
+        args: ["--profile=black"]
+
+  - repo: https://github.com/astral-sh/ruff-pre-commit
+    rev: v0.7.3
+    hooks:
+      - id: ruff
+        args: [--fix, "--extend-ignore", "E402"]
```

Comment:
Once this PR is tested and the imports are working as intended, we can enforce rule E402

Reply:
You can go ahead and enforce rule E402